### PR TITLE
Add HID Keyboard support to UefiHidDxe

### PR DESCRIPTION
**Pull request by joschock**
## Description

Adds HID keyboard support to UefiHidDxe input driver.

- [x] Impacts functionality?
  - Adds keyboard support.
- [ ] Impacts security?
- [ ] Breaking change?
- [ ] Includes tests?
- [x] Includes documentation?
  - standard rustdocs
## How This Was Tested

Tested with USB keyboard support in QEMU. Early iterations also tested in hardware. 

## Integration Instructions
Platforms will need to add `HiiKeyboardLayout = {path = "HidPkg/Crates/HiiKeyboardLayout"}` to the `[workspace.dependencies]` in their cargo.toml if not already present.

### CODE REVIEW

**Review by apop5 — APPROVED** [review]

**Comment by makubacki** [comment]
@joschock, please hold until I can come back and review on Monday.

**Comment by joschock** [comment]
> @joschock, please hold until I can come back and review on Monday.

Not in any rush, take your time.

File: HidPkg/UefiHidDxe/src/keyboard.rs [hunk 1]
```diff
@@ -8,38 +8,1026 @@
...
+  }
+
+  //Create the Keyboard Context and install the SimpleTextIn/SimpleTextInEx interfaces.
+  fn install_keyboard_interfaces(
+    self,
+    controller: efi::Handle,
+    hid_context: *mut HidContext,
+  ) -> Result<*mut KeyboardContext, efi::Status> {
+    // retrieve a reference to boot services.
+    // Safety: BOOT_SERVICES must ahve been initialized to point to the UEFI Boot Services table.
```
Comment (makubacki) [inline HidPkg/UefiHidDxe/src/keyboard.rs]: nit typo

```suggestion
    // Safety: BOOT_SERVICES must have been initialized to point to the UEFI Boot Services table.

```

**Review by makubacki — APPROVED** [review]

File: HidPkg/UefiHidDxe/src/keyboard.rs (hunk 1 above)
Comment (joschock) [reply HidPkg/UefiHidDxe/src/keyboard.rs]: I see you fixed it for me - perfect :). Thank you.

**Pull request merged**